src/metametrics/metrics/armorm_metric.py

```
@@ -0,0 +1,135 @@
+
+            assert torch.isclose(torch.sum(multi_obj_rewards * multi_obj_coeffs, dim=1), preference_score, atol=1e-3)
+            
+            return {attr: multi_obj_rewards[0][self.attribute_dict[attr]].item() for attr in scoring_attributes}
+
+    def __init__(self, model_id: str = "RLHFlow/ArmoRM-Llama3-8B-v0.1", scoring_attributes: Optional[List[str]] = None, trust_remote_code: bool = True, **kwargs):
+        self.model_id = model_id
+        self.trust_remote_code = validate_bool(trust_remote_code)
+
+        self.rm = self.ArmoRMPipeline(model_id=self.model_id, trust_remote_code=self.trust_remote_code)
+        if scoring_attributes is None:
+            self.scoring_attributes = [self.rm.attributes[1]]  # Default to ['helpsteer-correctness']
+        else:
+            # Validate all provided attributes exist
+            for attr in scoring_attributes:
+                if attr not in self.rm.attributes:
+                    raise ValueError(f"Invalid scoring attribute: {attr}")
+            self.scoring_attributes = scoring_attributes
+
+    def score(self, predictions: List[str], references: Union[None, List[List[str]]] = None, sources: Union[None, List[str]] = None) -> List[Dict[str, float]]:
+        if sources is None or predictions is None:
+            raise ValueError("Both sources (questions) and predictions (answers) must be provided.")
+
+        df = pd.DataFrame({"question": sources, "prediction": predictions})
+        scores = []
+
+        for _, row in tqdm(df.iterrows(), total=len(df)):
+            try:
+                score = self.rm(
+                    messages=[{"role": "user", "content": row['question']}, {"role": "assistant", "content": row['prediction']}],
+                    scoring_attributes=self.scoring_attributes
+                )
+                scores.append(score)
+            except Exception as e:
+                logger.error(f"Error computing score: {e}. Assigning default score of -99 for all attributes.")

```

If you are logging for an error, then you should raise an error.